src/main/java/hudson/remoting/AtmostOneThreadExecutor.java

```diff
@@ -71,10 +71,12 @@ public boolean isTerminated() {
 
     public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
         synchronized (q) {
-            long start = System.currentTimeMillis();
-            long end = start+unit.toMillis(timeout);
-            while (isAlive() && System.currentTimeMillis()<end) {
-                q.wait(end-System.currentTimeMillis());
+            long now = System.currentTimeMillis();
+            long end = now + unit.toMillis(timeout);
+            while (isAlive() && now < end) {
+                q.wait(end - now);
+                // XXX this is not safe against Clock skew - but System.nanoTime() has performance implications.
```

Comment:
Correction. I object to `XXX` I am fine with `FIXME` or `TODO` as
appropriate

On Wednesday, February 18, 2015, Jesse Glick <notifications@github.com>
wrote:

> In src/main/java/hudson/remoting/AtmostOneThreadExecutor.java
> <https://github.com/jenkinsci/remoting/pull/29#discussion_r24929965>:
>
> > @@ -71,10 +71,12 @@ public boolean isTerminated() {
> >
> >      public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
> >          synchronized (q) {
> > -            long start = System.currentTimeMillis();
> > -            long end = start+unit.toMillis(timeout);
> > -            while (isAlive() && System.currentTimeMillis()<end) {
> > -                q.wait(end-System.currentTimeMillis());
> > +            long now = System.currentTimeMillis();
> > +            long end = now + unit.toMillis(timeout);
> > +            while (isAlive() && now < end) {
> > +                q.wait(end - now);
> > +                // XXX this is not safe against Clock skew - but System.nanoTime() has performance implications.
>
> BTW at @stephenc <https://github.com/stephenc>’s insistence we use TODO
> for all such comments.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/remoting/pull/29/files#r24929965>.
>


-- 
Sent from my phone